.ci/snap_publish.sh

```diff
@@ -6,16 +6,28 @@ export SNAPCRAFT_BUILD_ENVIRONMENT_CPU
 SNAPCRAFT_BUILD_ENVIRONMENT_MEMORY=16G
 export SNAPCRAFT_BUILD_ENVIRONMENT_MEMORY
 
-# Prevent running if we've already published this revision
+# Install git to detect branch names / revisions
 apt update
 apt install git -y
-touch snap-last-run-hash
-if [ "$(cat snap-last-run-hash)" == "$(git rev-parse HEAD)-${RELEASE_CHANNEL}" ]; then
-  echo "Already built and published $(git rev-parse HEAD) on ${RELEASE_CHANNEL}. Nothing to do!";
+
+# Set release channel based on branch
+branch_name=$(git rev-parse --symbolic-full-name --abbrev-ref HEAD)
+if [[ ${branch_name} == "master" && ${RELEASE_CHANNEL} == "beta_or_edge" ]]; then
+  RELEASE_CHANNEL="edge"
+else
+  RELEASE_CHANNEL="beta"
+fi
```

Comment:
This is non-obvious on first read, why does not beta_or_edge potentially give me beta?

Reply:
Ah right, I'll add it as a comment. Basically: a build with `RELEASE_CHANNEL=beta_or_edge` only runs at night as a schedule (or if a schedule is manually triggered). The schedules target either `master` or a release branch (`1.0`, `1.2`, ..). If it's run on master it should go into Snap's 'edge' channel, otherwise it's a pre-release dot-version (i.e., latest released version + fixes accumulated in release branch) thus beta.

Reply:
Done!